src/MongoDB.Driver/ReplaceOptions.cs

```diff
@@ -55,6 +56,7 @@ internal static ReplaceOptions From(UpdateOptions updateOptions)
         // fields
         private bool? _bypassDocumentValidation;
         private Collation _collation;
+        private BsonValue _hint;
```

Comment:
You need to set `Hint` between lines 49-50 in the `From` method.

Reply:
Thanks! Didn't notice this because of region.
Done.